core/src/main/java/org/pentaho/di/core/xml/XMLCheck.java

```diff
@@ -0,0 +1,81 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2002-2017 by Hitachi Vantara : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.core.xml;
+
+import java.io.InputStream;
+
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+
+import org.apache.commons.vfs2.FileObject;
+import org.pentaho.di.core.exception.KettleException;
+import org.xml.sax.helpers.DefaultHandler;
+
+public class XMLCheck {
+
+  public static class XMLTreeHandler extends DefaultHandler {
+
+  }
+
+  /**
+   * Checks an xml file is well formed.
+   *
+   * @param file
+   *          The file to check
+   * @return true if the file is well formed.
+   */
+  public static final boolean isXMLFileWellFormed( FileObject file ) throws KettleException {
+    boolean retval = false;
+    try {
+      retval = isXMLWellFormed( file.getContent().getInputStream() );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *getInputStream*. Currently, there are execution paths that do not contain closure statements, for example, when SAXParserFactory.newSAXParser (in method isXMLWellFormed) throws an exception. Ensure the object returned by *getInputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).